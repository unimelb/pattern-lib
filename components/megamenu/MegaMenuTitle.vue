<template>
  <a
    :style="{ 'width': width && width || '50%' }"
    :aria-label="title"
    :href="href"
    class="page-title">{{ title }}</a>
</template>

<script>
export default {
  name: 'MegaMenuTitle',
  props: {
    title: {
      type: [String, Boolean],
      default: false,
    },
    href: {
      type: [String],
      required: true,
    },
    width: {
      type: [String, Boolean],
      default: false,
    },
  },
};
</script>
