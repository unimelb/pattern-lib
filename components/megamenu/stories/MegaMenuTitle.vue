<template>
  <MegaMenu
    :items="items"
    :top-menu="topMenu"
    active="/random-link"
    faculty-name="Faculty of Fine Arts and Music"
    faculty-link="/faculty-link"
    faculty-width="25.5%" />
</template>

<script>
import MegaMenu from '../MegaMenu.vue';
import MegaMenuDocs from './mega-menu-docs.md';
import sample from '../sample';

export default {
  components: { MegaMenu },
  readme: { custom: MegaMenuDocs, source: true, html: true },
  data() {
    return {
      items: [
        {
          title: 'Study',
          href: '/',
        },
        {
          title: 'Find a course',
          href: '/find-a-course',
        },
        {
          title: "Why's Melbourne",
          href: '/why-melbourne',
          feature: {
            title: 'Campaign title',
            text:
              'This should in theory be a super short para to sell the campaign, but I know <em>someone</em> will put too much text in',
            img: 'https://picsum.photos/id/1033/360/200.jpg',
            link: 'http://google.com',
          },
          items: [
            { title: 'Entry requirements1', href: '/random-link' },
            { title: 'Entry requirements2', href: 'http://www.google.com' },
            { title: 'Entry requirements3', href: 'http://www.google.com' },
            { title: 'Entry requirements4', href: 'http://www.google.com' },
            { title: 'Entry requirements5', href: 'http://www.google.com' },
          ],
        },
        {
          title: 'Admissions',
          href: '/admissions',
          feature: {
            title: 'Different campaign',
            img: 'https://picsum.photos/id/1040/360/200.jpg',
          },
          items: [
            { title: 'Entry non requirements6', href: 'http://www.google.com' },
            { title: 'Entry non requirements7', href: 'http://www.google.com' },
            {
              title: 'Entry non requirements8',
              href: 'http://www.google.com',
            },
            { title: 'Entry non requirements9', href: 'http://www.google.com' },
            {
              title: 'Entry non requirements10',
              href: 'http://www.google.com',
            },
            {
              title: 'Entry non requirements11',
              href: 'http://www.google.com',
            },
            {
              title: 'Entry non requirements12',
              href: 'http://www.google.com',
            },
            {
              title: 'Entry non requirements13',
              href: 'http://www.google.com',
            },
            {
              title: 'Entry non requirements14',
              href: 'http://www.google.com',
            },
            { title: 'Entry non requirements15', href: 'http://www.google.com' },
          ],
        },
        {
          title: 'Student experience',
          href: '/student-experience',
          feature: {
            alt: 'Screen reader only caption!',
            img: 'https://picsum.photos/id/1040/360/200.jpg',
            link: 'http://google.com',
          },
          items: [
            { title: 'Entry requirements21', href: '/2random-link' },
            { title: 'Entry requirements22', href: 'http://2www.google.com' },
            { title: 'Entry requirements23', href: 'http://2www.google.com' },
            { title: 'Entry requirements24', href: 'http://2www.google.com' },
            { title: 'Entry requirements25', href: 'http://2www.google.com' },
          ],
        },
        {
          title: 'Connect',
          href: '/connect',
        },
      ],
      topMenu: sample.topMenu,
    };
  },
};
</script>
