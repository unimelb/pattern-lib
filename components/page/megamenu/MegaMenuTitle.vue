<template>
  <a
    :style="{'width': width && width || '50%'}"
    :aria-label="title"
    :href="href"
    class="page-title"
    tabindex="2">{{ title }}</a>
</template>

<script>
export default {
  name: 'MegaMenuTitle',
  props: {
    title: {
      type: [String, Boolean],
      default: false,
    },
    href: {
      type: [String],
      required: true,
    },
    width: {
      type: [String, Boolean],
      default: false,
    },
  },
};
</script>
