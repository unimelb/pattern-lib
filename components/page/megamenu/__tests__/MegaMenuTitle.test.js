import { shallow } from 'vue-test-utils';
import { toHaveNoViolations } from 'jest-axe';
import MegaMenuTitle from '../MegaMenuTitle.vue';

expect.extend(toHaveNoViolations);

describe('MegaMenuTitle', () => {
  it('should match snapshot', () => {
    const result = shallow(MegaMenuTitle).element;
    expect(result).toMatchSnapshot();
  });

  it('should have default props and correct types title/href/width/hasAperture/apertureColor', () => {
    const wrapper = shallow(MegaMenuTitle);
    const {
      title, href, width, hasAperture, apertureColor,
    } = wrapper.vm.$options.props;

    expect(title.type).toEqual([String, Boolean]);
    expect(wrapper.props().title).toBe(false);

    expect(href.type).toEqual([String]);
    expect(wrapper.props().href).toBe(undefined);

    expect(width.type).toEqual([String, Boolean]);
    expect(wrapper.props().width).toBe(false);

    expect(hasAperture.type).toBe(Boolean);
    expect(wrapper.props().hasAperture).toBe(false);
    expect(wrapper.find('.page-title__aperture').exists()).toBe(false);

    expect(apertureColor.type).toBe(String);
    expect(wrapper.props().apertureColor).toBe('black');
  });

  it('should render title from prop with correct type', () => {
    const title = 'Test name';
    const wrapper = shallow(MegaMenuTitle, {
      propsData: {
        title,
      },
    });

    expect(typeof wrapper.props().title).toBe('string');
    expect(wrapper.props().title).toBe(title);
    expect(wrapper.find('.page-title__text').text()).toBe(title);
  });

  it('should have title as link', () => {
    const wrapper = shallow(MegaMenuTitle);
    expect(wrapper.find('.page-title__text').element.tagName).toBe('A');
  });

  it('should render href from prop with correct type', () => {
    const href = '/';
    const wrapper = shallow(MegaMenuTitle, {
      propsData: {
        href,
      },
    });

    expect(typeof wrapper.props().href).toBe('string');
    expect(wrapper.props().href).toBe(href);
    expect(wrapper.find('.page-title__text').attributes().href).toBe(href);
  });

  it('should render with correct width from prop', () => {
    const width = '50px';
    const wrapper = shallow(MegaMenuTitle, {
      propsData: {
        width,
      },
    });

    expect(typeof wrapper.props().width).toBe('string');
    expect(wrapper.props().width).toBe(width);
    expect(wrapper.element.style.width).toBe(width);
  });

  it('should render with default width', () => {
    const defaultWidth = '50%';
    const wrapper = shallow(MegaMenuTitle);

    expect(wrapper.element.style.width).toBe(defaultWidth);
  });

  it('should render aperture', () => {
    const wrapper = shallow(MegaMenuTitle, {
      propsData: {
        hasAperture: true,
      },
    });

    expect(wrapper.find('.page-title__aperture').exists()).toBe(true);
  });

  it('should change aperture color', () => {
    const wrapper = shallow(MegaMenuTitle, {
      propsData: {
        hasAperture: true,
        apertureColor: 'red',
      },
    });
    expect(wrapper.find('.page-title__aperture').attributes().style).toBe('background: red;');
  });
});
