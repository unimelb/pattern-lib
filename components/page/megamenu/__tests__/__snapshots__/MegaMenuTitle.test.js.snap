// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MegaMenuTitle should match snapshot 1`] = `
<div
  class="page-title"
  style="width: 50%;"
>
  <div
    class="page-title__separator"
  />
   
  <a
    class="page-title__text"
  >
    
    false
  
  </a>
</div>
`;
